internal/server/data/schema.sql

```diff
@@ -220,6 +238,15 @@ CREATE TABLE providers (
     organization_id bigint
 );
 
+CREATE VIEW resolved_identity_groups AS
```

Comment:
Converted this complicated join to a view, this way our code for reading and writing remains simple. The complexity is in this single group joining view.